System/Console/Haskeline/Completion.hs

```diff
@@ -188,3 +189,14 @@ fixPath ('~':c:path) | isPathSeparator c = do
     home <- getHomeDirectory
     return (home </> path)
 fixPath path = return path
+
+fallbackCompletion :: Monad m => CompletionFunc m -> CompletionFunc m -> CompletionFunc m
```

Comment:
Can you add a Haddock explaining the semantics of this?  I'm not sure I understand the intended use.  In particular I find it problematic that it's possible for this to throw `error` which would crash the calling program.

In your use case, where there are two sets of possible arguments to `add`, would it make more sense to present *both* sets of options (module names and paths), i.e., concatenate the set of possible completions, rather than only displaying one or the other?

Reply:
Yeah I think the more obvious behavior would be to concatenate the set of completions, so maybe `fallbackCompletion` isn't such a good name.

I'll also remove the error, although I think this suggests that the unused text should be represented with just a length rather than a string which isn't guaranteed to have any relation to the actual input.